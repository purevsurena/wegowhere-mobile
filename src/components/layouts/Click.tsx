import React from 'react';
import RNBounceable from '@freakycoder/react-native-bounceable';

import { ClickProps, ClickDefaultProps } from '@/types/layouts';

function Click(props: ClickProps): JSX.Element {
  const { onPress, disabled, children, style, testID } = props;

  return (
    <RNBounceable
      testID={`${testID}`}
      onPress={onPress}
      disabled={disabled}
      style={{ ...style }}>
      {children}
    </RNBounceable>
  );
}

Click.defaultProps = ClickDefaultProps;

export default Click;
