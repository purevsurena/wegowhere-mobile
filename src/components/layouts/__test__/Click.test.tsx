import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import Click from '@/components/layouts/Click';
import { Text } from 'react-native';

// Mock the RNBounceable module
jest.mock('@freakycoder/react-native-bounceable', () => 'RNBounceable');

describe('<Click />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Click onPress={() => {}} children={<Text>Click Me</Text>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles press event', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <Click onPress={onPressMock} testID="click-component">
        <Text>Click Me</Text>
      </Click>,
    );

    fireEvent.press(getByTestId('click-component'));

    expect(onPressMock).toHaveBeenCalled();
  });

  it('renders with disabled state', () => {
    const { getByTestId } = render(
      <Click onPress={() => {}} disabled testID="click-component">
        <Text>Click Me</Text>
      </Click>,
    );

    fireEvent.press(getByTestId('click-component'));
  });

  it('matches snapshot with custom props', () => {
    const tree = renderer
      .create(
        <Click onPress={() => {}} disabled testID="click-component">
          <Text>Click Me</Text>
        </Click>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
