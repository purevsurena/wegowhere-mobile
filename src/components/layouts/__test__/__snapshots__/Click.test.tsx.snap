// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Click /> matches snapshot with custom props 1`] = `
<RNBounceable
  disabled={true}
  onPress={[Function]}
  style={{}}
  testID="click-component"
>
  <Text>
    Click Me
  </Text>
</RNBounceable>
`;

exports[`<Click /> renders correctly 1`] = `
<RNBounceable
  disabled={false}
  onPress={[Function]}
  style={{}}
  testID="undefined"
>
  <Text>
    Click Me
  </Text>
</RNBounceable>
`;
